scaleway/resource_compute_instance_security_group.go

```diff
@@ -0,0 +1,368 @@
+						},
+					},
+				},
+			},
+			"zone":       zoneSchema(),
+			"project_id": projectIDSchema(),
+		},
+	}
+}
+
+func resourceScalewayComputeInstanceSecurityGroupCreate(d *schema.ResourceData, m interface{}) error {
+	meta := m.(*Meta)
+	instanceApi := instance.NewAPI(meta.scwClient)
+
+	zone, err := getZone(d, meta)
+	if err != nil {
+		return err
+	}
+
+	projectID, err := getProjectId(d, meta)
+	if err != nil {
+		return err
+	}
+
+	name := d.Get("name").(string)
+	if name == "" {
+		name = getRandomName("sg")
+	}
+
+	res, err := instanceApi.CreateSecurityGroup(&instance.CreateSecurityGroupRequest{
+		Name:                  name,
+		Zone:                  zone,
+		Organization:          projectID,
+		Description:           d.Get("description").(string),
+		Stateful:              true,
+		InboundDefaultPolicy:  instance.SecurityGroupPolicy(d.Get("inbound_default_policy").(string)),
+		OutboundDefaultPolicy: instance.SecurityGroupPolicy(d.Get("outbound_default_policy").(string)),
+	})
+	if err != nil {
+		return err
+	}
+
+	d.SetId(newZonedId(zone, res.SecurityGroup.ID))
+	return resourceScalewayComputeInstanceSecurityGroupUpdate(d, m)
+}
+
+func resourceScalewayComputeInstanceSecurityGroupRead(d *schema.ResourceData, m interface{}) error {
+	meta := m.(*Meta)
+	instanceApi := instance.NewAPI(meta.scwClient)
+
+	zone, ID, err := parseZonedID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	res, err := instanceApi.GetSecurityGroup(&instance.GetSecurityGroupRequest{
+		SecurityGroupID: ID,
+		Zone:            zone,
+	})
+	if err != nil {
+		if is404Error(err) {
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+
+	d.Set("zone", zone)
+	d.Set("project_id", res.SecurityGroup.Organization)
+	d.Set("name", res.SecurityGroup.Name)
+	d.Set("description", res.SecurityGroup.Description)
+	d.Set("inbound_default_policy", res.SecurityGroup.InboundDefaultPolicy.String())
+	d.Set("outbound_default_policy", res.SecurityGroup.OutboundDefaultPolicy.String())
+
+	resRules, err := instanceApi.ListSecurityGroupRules(&instance.ListSecurityGroupRulesRequest{
+		Zone:            zone,
+		SecurityGroupID: ID,
+	}, scw.WithAllPages())
+	if err != nil {
+		return err
+	}
+
+	inboundRules := ([]interface{})(nil)
+	for _, rule := range resRules.Rules {
+		if rule.Direction == instance.SecurityGroupRuleDirectionOutbound {
+			continue
+		}
+		inboundRules = append(inboundRules, securityGroupRuleFlatten(rule))
+	}
+	d.Set("inbound_rule", inboundRules)
+
+	return nil
+}
+
+func portRangeFlatten(from, to uint32) string {
+	if to == 0 {
+		to = from
+	}
+	return fmt.Sprintf("%d-%d", from, to)
+}
+
+func portRangeFormat(i interface{}) string {
+	return portRangeFlatten(portRangeExpand(i))
+}
+
+func ipv4RangeFormat(i interface{}) string {
+	ipRange := i.(string)
+	if !strings.Contains(ipRange, "/") {
+		ipRange = ipRange + "/32"
+	}
+	return ipRange
+}
+
+func portRangeExpand(i interface{}) (uint32, uint32) {
+	portRange := i.(string)
+
+	var from, to uint32
```

Comment:
[Switch to uint16](https://play.golang.org/p/DPW4np-0lnz)

Reply:
API uses uint32